src/NetscriptJSEvaluator.ts

```diff
@@ -76,15 +77,30 @@ function addDependencyInfo(script: Script, seenStack: Script[]) {
  * @param scripts array of other scripts on the server
  * @param seenStack A stack of scripts that were higher up in the import tree in a recursive call.
  */
-function generateLoadedModule(script: Script, scripts: Map<ScriptFilePath, Script>, seenStack: Script[]): LoadedModule {
+async function generateLoadedModule(
```

Comment:
I need to think carefully about the implications of this being async. There's a lot of subtle stuff tied up in this.